api/src/test/java/org/pmiops/workbench/auth/DelegatedUserCredentialsTest.java

```diff
@@ -138,11 +143,23 @@ public void testRefreshFlow() throws IOException {
                   .build();
             });
     // Register the expected service account & access token with the mock token server transport.
-    mockTokenServerTransport.addServiceAccount(SERVICE_ACCOUNT_EMAIL, "access-token");
+    mockTokenServerTransport.addServiceAccount(SERVICE_ACCOUNT_EMAIL, MOCK_ACCESS_TOKEN);
 
     // Kick off the refresh flow.
-    creds.refresh();
-
-    assertThat(creds.getAccessToken().getTokenValue()).isEqualTo("access-token");
+    delegatedCredentials.refresh();
+
+    // Verify the call to IAM Credentials API.
+    ArgumentCaptor<SignJwtRequest> captor = ArgumentCaptor.forClass(SignJwtRequest.class);
+    verify(mockIamCredentialsClient, Mockito.times(1)).signJwt(captor.capture());
+    assertThat(captor.getValue().getName())
+        .isEqualTo("projects/-/serviceAccounts/" + SERVICE_ACCOUNT_EMAIL);
+
+    // The mockTokenServerTransport class runs some lightweight verification of its own (i.e.,
+    // ensuring the signed JWT can be parsed and that the service account is known). Beyond that,
+    // we mainly care that the access token is returned and has a correct expiration.
+    assertThat(delegatedCredentials.getAccessToken().getTokenValue()).isEqualTo(MOCK_ACCESS_TOKEN);
+    assertThat(
+            delegatedCredentials.getAccessToken().getExpirationTime().toInstant().getEpochSecond())
```

Comment:
I'd declare a static `Instant` so you have the same calculation every time. May need to inject a fake clock somewhere though.